website/orders/tests/test_models.py

```diff
@@ -0,0 +1,558 @@
+import datetime
+from datetime import timedelta
+from unittest.mock import patch, MagicMock
+from freezegun import freeze_time
+
+import pytz
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.test import TestCase, override_settings
+from django.utils import timezone
+from django.conf import settings
+from guardian.shortcuts import assign_perm
+
+from orders import models
+from venues.models import Venue
+from users.models import User as User_models
+
+
+User = get_user_model()
+
+
+class OrderModelTests(TestCase):
+
+    fixtures = ["venues.json", "users.json"]
+
+    def setUp(self):
+        self.normal_user = User.objects.get(pk=2)
+        self.admin_user = User.objects.get(pk=1)
+        venue_pk_1 = Venue.objects.get(pk=1)
+        self.order_venue = models.OrderVenue.objects.create(venue=venue_pk_1)
+        self.product = models.Product.objects.create(name="Test product", current_price=1.25)
+        self.product.available_at.add(self.order_venue)
+        self.product.save()
+        self.product_2 = models.Product.objects.create(name="Test product 2", current_price=2.00)
+        self.product_2.available_at.add(self.order_venue)
+        self.product_2.save()
+        self.shift = models.Shift.objects.create(
+            venue=self.order_venue, start_date=timezone.now(), end_date=timezone.now() + timedelta(hours=4)
+        )
+
+    def test_validate_barcode(self):
+        def barcode_wrong_checksum():
+            models.validate_barcode("8710624356915")
+
+        def barcode_too_short():
+            models.validate_barcode("87106243569")
+
+        def barcode_not_only_digits():
+            models.validate_barcode("87C062B356A")
+
+        models.validate_barcode("8710624356910")
+        models.validate_barcode("5449000000439")
+        models.validate_barcode("61940017")
+        models.validate_barcode("87654325")
+        models.validate_barcode(None)
+        self.assertRaises(ValidationError, barcode_wrong_checksum)
```

Comment:
```suggestion
with self.subTest("Wrong checksum"):
        self.assertRaises(ValidationError, models.validate_barcode, "8710624356915")
```
and same for the others